usr/src/uts/common/fs/zfs/zil.c

```diff
@@ -1057,12 +1061,13 @@ zil_lwb_write_start(zilog_t *zilog, lwb_t *lwb)
 static lwb_t *
 zil_lwb_commit(zilog_t *zilog, itx_t *itx, lwb_t *lwb)
 {
-	lr_t *lrc = &itx->itx_lr; /* common log record */
-	lr_write_t *lrw = (lr_write_t *)lrc;
+	lr_t *lrcb, *lrc = &itx->itx_lr; /* common log record */
```

Comment:
I would either split the declaration lines here so that there is one variable per declaration or I would move the initializations to separate lines. It is harder to read the code when there are both initialized and uninitialized variables in the same declarations.
Also, maybe it would make sense to add your description of the variables as a comment in the code.